sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -292,13 +345,29 @@ private void exportBatches(ArrayList<ReadableSpan> spanList) {
       return Collections.unmodifiableList(spanDataBuffer);
     }
 
-    // Exports the list of Span protos to all the ServiceHandlers.
-    private void onBatchExport(List<SpanData> spans) {
-      // In case of any exception thrown by the service handlers continue to run.
+    // Exports the list of SpanData to the SpanExporter.
+    private void onBatchExport(final List<SpanData> spans) {
+      Future<?> submission =
+          executorService.submit(
+              new Runnable() {
+                @Override
+                public void run() {
+                  // In case of any exception thrown by the service handlers catch and log.
+                  try {
+                    spanExporter.export(spans);
+                  } catch (Throwable t) {
+                    logger.log(Level.WARNING, "Exception thrown by the export.", t);
+                  }
+                }
+              });
       try {
-        spanExporter.export(spans);
-      } catch (Throwable t) {
-        logger.log(Level.WARNING, "Exception thrown by the export.", t);
+        // wait at most for the configured timeout.
+        submission.get(exporterTimeoutMillis, TimeUnit.MILLISECONDS);
```

Comment:
Can we avoid blocking the worker thread here?

Reply:
I think the only approach would be to submit another scheduled job that would do this work. If we're ok with a second monitor thread to kill the other Future, it would be a pretty easy addition to make.

Reply:
Let's just file an issue for the moment to track this and document what options we have.

Reply:
#750